libs/portal-integration-angular/src/lib/core/components/page-header/page-header.component.spec.ts

```diff
@@ -104,76 +98,153 @@ describe('PageHeaderComponent', () => {
+      {
+        label: 'Venue',
+        value: 'AIE Munich',
+      },
+      {
+        label: 'Status',
+        value: 'Confirmed',
+      },
+    ]
+    expect((await pageHeaderHarness.getObjectInfos()).length).toEqual(0)
+
+    component.objectDetails = objectDetailsWithoutIcons
+    
+    expect((await pageHeaderHarness.getObjectInfos()).length).toEqual(2)
+    const objectDetailLabels = await pageHeaderHarness.getObjectDetailLabels()
+    const objectDetailValues = await pageHeaderHarness.getObjectDetailValues()
+    const objectDetailIcons = await pageHeaderHarness.getObjectDetailIcons()
+    expect(objectDetailLabels.length).toEqual(2)
+    expect(objectDetailValues.length).toEqual(2)
+    expect(objectDetailIcons.length).toEqual(0)
+
+    objectDetailLabels.forEach(async (label, i) => {
+      expect(await label.text()).toEqual(objectDetailsWithoutIcons[i].label)
+    })
+
+    objectDetailValues.forEach(async (value, i) => {
+      expect(await value.text()).toEqual(objectDetailsWithoutIcons[i].value)
+    })
+  })
+
+  it("should render objectDetails with icons as object info in the page header", async () => {
+    const objectDetailsWithIcons = [
+      {
+        label: 'Venue',
+        value: 'AIE Munich',
+      },
+      {
+        label: 'Status',
+        value: 'Confirmed',
+        icon: PrimeIcons.CHECK
+      },
+      {
+        label: 'Done?',
+        icon: PrimeIcons.EXCLAMATION_CIRCLE
+      },
+      {
+        label: 'Empty'
+      }
+    ]
+    expect((await pageHeaderHarness.getObjectInfos()).length).toEqual(0)
+
+    component.objectDetails = objectDetailsWithIcons
+    
+    expect((await pageHeaderHarness.getObjectInfos()).length).toEqual(4)
+    const objectDetailLabels = await pageHeaderHarness.getObjectDetailLabels()
+    const objectDetailValues = await pageHeaderHarness.getObjectDetailValues()
+    const objectDetailIcons = await pageHeaderHarness.getObjectDetailIcons()
+    expect(objectDetailLabels.length).toEqual(4)
+    expect(objectDetailValues.length).toEqual(3)
+    expect(objectDetailIcons.length).toEqual(2)
+
+    objectDetailLabels.forEach(async (label, i) => {
+      expect(await label.text()).toEqual(objectDetailsWithIcons[i].label)
+    })
+
+    objectDetailValues.forEach(async (value, i) => {
+      if(objectDetailsWithIcons[i].value) {
+        expect(await value.text()).toEqual(objectDetailsWithIcons[i].value)
+      }
+    })
+
+    expect(await objectDetailIcons[0].getAttribute('class')).toEqual(objectDetailsWithIcons[1].icon)
+    expect(await objectDetailIcons[1].getAttribute('class')).toEqual(objectDetailsWithIcons[2].icon)
   })
 
   it('should show overflow actions when menu overflow button clicked', () => {
     component.actions = mockActions
     fixture.detectChanges()
 
     const menuOverflowButton = fixture.debugElement.nativeElement.querySelector(
-      '[data-testid="ocx-page-header-overflow-action-button"]'
+      '[name="ocx-page-header-overflow-action-button"]'
```

Comment:
Why are you not using the Harness in this test?

Reply:
Good catch seems like I overlooked one of the older tests. I'll fix it.